File: src/main/java/com/pubnub/api/endpoints/channel_groups/ChannelGroupService.java
package com.pubnub.api.endpoints.channel_groups;

import com.pubnub.api.models.server.Envelope;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;

import java.util.Map;


public interface ChannelGroupService {
    @GET("v1/channel-registration/sub-key/{subKey}/channel-group")
    Call<Envelope<Object>> listAllChannelGroup(@Path("subKey") String subKey,
                                               @QueryMap Map<String, String> options);

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}")
    Call<Envelope<Object>> allChannelsChannelGroup(@Path("subKey") String subKey,
                                                   @Path("group") String group,
                                                   @QueryMap Map<String, String> options);

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}")
    Call<Envelope> addChannelChannelGroup(@Path("subKey") String subKey,
                                          @Path("group") String group,
                                          @QueryMap Map<String, String> options);

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}")
    Call<Envelope> removeChannel(@Path("subKey") String subKey,
                                 @Path("group") String group,
                                 @QueryMap Map<String, String> options);

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}/remove")
    Call<Envelope> deleteChannelGroup(@Path("subKey") String subKey,
                                      @Path("group") String group,
                                      @QueryMap Map<String, String> options);
}

File: src/main/kotlin/com/pubnub/api/services/ChannelGroupService.kt
package com.pubnub.api.services

import com.pubnub.api.models.server.Envelope
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.QueryMap

internal interface ChannelGroupService {

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group")
    fun listAllChannelGroup(
        @Path("subKey") subKey: String,
        @QueryMap options: Map<String, String>
    ): Call<Envelope<Map<String, Any>>>

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}")
    fun allChannelsChannelGroup(
        @Path("subKey") subKey: String,
        @Path("group") group: String,
        @QueryMap options: Map<String, String>
    ): Call<Envelope<Map<String, Any>>>

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}")
    fun addChannelChannelGroup(
        @Path("subKey") subKey: String,
        @Path("group") group: String,
        @QueryMap options: Map<String, String>
    ): Call<Void>

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}")
    fun removeChannel(
        @Path("subKey") subKey: String,
        @Path("group") group: String,
        @QueryMap options: Map<String, String>
    ): Call<Void>

    @GET("v1/channel-registration/sub-key/{subKey}/channel-group/{group}/remove")
    fun deleteChannelGroup(
        @Path("subKey") subKey: String,
        @Path("group") group: String,
        @QueryMap options: Map<String, String>
    ): Call<Void>
}
